spec/features/create_folio_object_spec.rb

```diff
@@ -0,0 +1,74 @@
+
+RSpec.describe 'Use Argo to create an item object with a folio instance HRID' do
+  let(:random_word) { random_phrase }
+  let(:object_label) { "Object Label for #{random_word}" }
+  let(:start_url) { "#{Settings.argo_url}/registration" }
+  let(:source_id) { "create-obj-folio-instance-hrid-test:#{random_alpha}" }
+  let(:folio_instance_hrid) { Settings.test_folio_instance_hrid }
+  let(:catalog_object_label) { 'The means to prosperity' } # will be pulled from folio
+  let(:folio_instance_hrid_updated) { 'a123' }
+  let(:catalog_object_label_updated) { 'A la francaise' } # the updated label after we change the hrid
+  let(:user_tag) { 'Some : UniqueTagValue' }
+  let(:project) { 'Awesome Folio Project' }
+
+  before do
+    authenticate!(start_url:,
+                  expected_text: 'Register DOR Items')
+  end
+
+  scenario do
+    abort 'SKIPPING: Folio not enabled' unless Settings.folio.enabled
+
+    # fill in registration form
+    select 'integration-testing', from: 'Admin Policy'
+    select 'integration-testing', from: 'Collection'
+    select 'book', from: 'Content Type'
+    fill_in 'Tag', with: user_tag
+    fill_in 'Project Name', with: project
+
+    fill_in 'Source ID', with: source_id
+    fill_in 'Folio Instance HRID', with: folio_instance_hrid
+    fill_in 'Label', with: object_label # will be overwritten and checked below
+
+    click_button 'Register'
+
+    # wait for object to be registered
+    expect(page).to have_text 'Items successfully registered.'
+
+    bare_object_druid = find('table a').text
+    object_druid = "druid:#{bare_object_druid}"
+    puts " *** create folio object spec druid: #{object_druid} ***" # useful for debugging
+
+    visit "#{Settings.argo_url}/view/#{object_druid}"
+
+    # wait for registrationWF to finish
+    reload_page_until_timeout!(text: 'v1 Registered')
+
+    # look for metadata
+    expect(page).to have_text(user_tag)
+    expect(page).to have_text("Project : #{project}")
+    expect(page).to have_text(Settings.test_folio_instance_hrid)
+    expect(page).to have_text(catalog_object_label) # this was pulled from folio, overwriting used entered label
+    expect(page).to have_text("Registered By : #{AuthenticationHelpers.username}")
+
+    # edit folio_instance_hrid
+    click_link 'Manage Folio Instance HRID'
+    fill_in 'catalog_record_id_catalog_record_ids_attributes_0_value', with: folio_instance_hrid_updated
+    click_button 'Update'
+
+    # look for updated hrid and refresh metadata
+    expect(page).to have_text(folio_instance_hrid_updated)
+    click_button 'Manage description'
+    click_link 'Refresh'
+    reload_page_until_timeout!(text: catalog_object_label_updated) # updated label pulled from folio for new HRID
+
+    # look for metadata source facet
+    visit Settings.argo_url
+    click_link 'Show more facets'
+    click_button 'Metadata Source (Multi)'
+    within('.blacklight-metadata_source_ssim') do
+      expect(page).to have_text('Folio')
+    end
```

Comment:
it would be nice to check the total number of objects shown in this facet and see if it increased by one after we registered this object, but I can't think of a good way of finding this number (there isn't an obvious selector to use)

Reply:
Check out the `access_indexing_spec` for testing indexing via a facet.

Reply:
Ok, so basically search for the single druid and see if the relevant facet value is 1.  Thanks.  I replaced this part of the test with that check.